ledger-core/inc/core/utils/DerivationPath.hpp

```diff
@@ -34,35 +34,42 @@
 #include <string>
 #include <vector>
 
+#include <core/api/DerivationPath.hpp>
 #include <core/utils/Exception.hpp>
 
 namespace ledger {
     namespace core {
-        class DerivationPath {
+        class DerivationPath : public api::DerivationPath {
         public:
             explicit DerivationPath(const std::string& path);
-            explicit DerivationPath(const std::vector<uint32_t>& path);
+            explicit DerivationPath(const std::vector<int32_t>& path);
```

Comment:
Even though I get why you need to change the type, it’s a regression here, because we would authorize `m/44'/-1'/0` for instance.

The fix you need prior to this is to add the support of unsigned integrals to Djinni.